proxyd/backend.go

```diff
@@ -921,22 +948,42 @@ func (bg *BackendGroup) ProcessMulticallResponses(ch chan *multicallTuple, ctx c
 		resp := multicallResp.response
 		backendName := multicallResp.backendName
 
+		// if failed response, store it
 		if resp.error != nil {
-			log.Error("received error response from multicall channel",
-				"req_id", GetReqID(ctx),
-				"auth", GetAuthCtx(ctx),
-				"err", resp.error,
-				"backend", backendName,
-			)
-			finalResp = resp
+			errResp = resp
 			continue
 		}
+
+		// success or rpc error response
+		if multicallRPCErrorCheck {
+			// check if any of the response is an RPC error
+			var rpcError = false
+			for _, rpcRes := range resp.RPCRes {
+				// check if the ID is already in the success response
```

Comment:
I think ID should be unique given it's part of the `handleBatchRPC` flow. It's either the incoming message is a batch request (which should have unique IDs otherwise they wouldn't know how to process the response), or a single request.
@jelias2 would be good if you could help to confirm.
cc: @danyalprout

Reply:
Theres one key thing is that currently we don't allow batch requests in multicall at this point, so all IDs will be most likely 1

```
	if bg.GetRoutingStrategy() == MulticallRoutingStrategy && isValidMulticallTx(rpcReqs) && !isBatch {
		backendResp := bg.ExecuteMulticall(ctx, rpcReqs)
		return backendResp.RPCRes, backendResp.ServedBy, backendResp.error
	}

```